lang_python.adoc

```diff
@@ -103,9 +129,14 @@ available out of the box in standard libraries since Python v3.4, or
 since v2.4 using https://pypi.python.org/pypi/enum34[enum34] PyPi
 compatibility layer.
 
-* On Debian / Ubuntu, this library can be installed with `sudo apt-get
-  install python-enum34`
-* Otherwise, one can use just `sudo pip install enum34`
+* One usually doesn't need to install this lib manually,
+   `pip` will install it automatically if it is needed.
+* One can also install it manually:
+** Otherwise, one can use just `sudo pip install enum34`. Don't do it on later pythons!
+** On Debian / Ubuntu, this library can be installed with `sudo apt-get install python-enum34`
+* If you have to use 3.4 you may want to replace `enum.py` with the one from
+   https://raw.githubusercontent.com/python/cpython/3.6/Lib/enum.py[python 3.6]
+   in order to get `IntFlag` (not currently used in KS, but may be used in future).
```

Comment:
This is bad advice - you really shouldn't be replacing files in your Python installation's stdlib. If you need a newer `enum` module and for some reason can't upgrade to 3.6, you should copy the new `enum` module to `site-packages` (or some other directory on `sys.path`), not into the stdlib directory.

In any case, considering that KS currently doesn't use any 3.6 `enum` features, I think this advice shouldn't be added to the documentation yet.

Reply:
>This is bad advice - you really shouldn't be replacing files in your Python installation's stdlib.

Why not? it works.

>In any case, considering that KS currently doesn't use any 3.6 enum features, I think this advice shouldn't be added to the documentation yet.

OK

Reply:
> Why not? it works.

Because you're replacing the existing standard `enum.py`, so there's no good way to revert the change without reinstalling Python. Putting the file into `site-packages` is also not *great* (because then you have a file in `site-packages` without any package metadata, so pip won't know what to do with it or how to uninstall it), but it's still better than replacing the original `enum.py`.

Reply:
>Because you're replacing the existing standard enum.py, so there's no good way to revert the change without reinstalling Python.

If one considers he may want to revert, he can keep a backup. I have kept a backup for the case of 3.6 -> 3.4, but everything works fine.

Of course in an ideal case the standard lib of python is just a set of preinstalled packages that can be upgraded independently of python. But it hasn't been done yet, brobably they consider it as not worthing the effort :(.